src/pages/token/Swap.tsx

```diff
@@ -0,0 +1,484 @@
+              onBlur={() => setIsToInputFocused(false)}
+            />
+            <TokenLogo token={otherToken(inputToken)} />
+          </div>
+        </$InputContainer>
+      </div>
+
+      {onboardingState !== OnboardingState.AccountConnected ? (
+        <OnboardingTriggerButton size={ButtonSize.BasePlus} />
+      ) : error ? (
+        <div tw="flex h-3 justify-center rounded-0.75 border border-solid border-color-layer-4 p-0.75">
+          <div tw="flex items-center gap-0.5 leading-5">
+            <WarningFilled tw="h-[15.6px] w-[17.3px] text-red" />
+            <div tw="text-base text-color-text-0">{capitalize(error.message)}</div>
+          </div>
+        </div>
+      ) : (
+        <Button
+          tw="h-3 p-0.75"
+          state={
+            !quote || !hasSufficientBalance
+              ? ButtonState.Disabled
+              : hasPendingSwap
+                ? ButtonState.Loading
+                : ButtonState.Default
+          }
+          disabled={!quote || !hasSufficientBalance || hasPendingSwap}
+          onClick={onSwap}
+          action={ButtonAction.Primary}
+        >
+          <div tw="text-base leading-5">
+            {hasSufficientBalance
+              ? stringGetter({ key: STRING_KEYS.SWAP })
+              : stringGetter({ key: STRING_KEYS.INSUFFICIENT_BALANCE })}
+          </div>
+        </Button>
+      )}
+
+      <$Accordion
+        triggerRotation={180}
+        triggerIcon={
+          <div tw="flex items-center">
+            <CaretDown tw="h-0.375 text-color-text-0" />
+          </div>
+        }
+        items={[
+          {
+            header: (
+              <ExchangeRate
+                gas={gas}
+                selectedToken={inputToken}
+                usdcPerDydx={usdcPerDydx}
+                priceImpact={priceImpact}
+              />
+            ),
+            content: (
+              <QuoteDetails isLoading={isLoading || isPlaceholderData} priceImpact={priceImpact} />
+            ),
+          },
+        ]}
+      />
+    </div>
+  );
+};
+
+const QuoteDetails = ({ priceImpact, isLoading }: { priceImpact?: number; isLoading: boolean }) => {
+  const stringGetter = useStringGetter();
+
+  return (
+    <div
+      tw="flex flex-col gap-0.5 py-0.625 leading-5 text-color-text-0 font-small-medium"
+      style={{ opacity: isLoading ? 0.6 : 1 }}
+    >
+      <div tw="flex items-center justify-between gap-0.5">
+        <div tw="flex items-center gap-0.375">
+          <div>{stringGetter({ key: STRING_KEYS.PRICE_IMPACT })}</div>
+          {/* TODO: add copy for price impact helper text */}
+          <WithTooltip tooltip="price-impact">
+            <Icon iconName={IconName.HelpCircle} tw="h-0.625 w-0.625" />
+          </WithTooltip>
+        </div>
+        <div>
+          {priceImpact ? (
+            <Output tw="inline" value={priceImpact / 100} type={OutputType.Percent} />
+          ) : (
+            '-'
+          )}
```

Comment:
`<Output>` component is smart enough to display `-` if we pass it null.

`value={priceImpact ? priceImpact / 100 : null}` I think we should also add an isLoading to the `<Output>`

```
<Output
  css={{
    display: isLoading ? 'flex' : 'inline',
  }}
  isLoading={isLoading}
  value={priceImpact ? priceImpact / 100 : null}
  type={OutputType.Percent}
/>
```